core/src/setup/Kokkos_Setup_SYCL.hpp

```diff
@@ -41,4 +41,6 @@
   } while (0)
 #endif
 
+using sycl::ext::oneapi::experimental::printf;
```

Comment:
I do not think it is acceptable to import into the global namespace.
Either we provide a `Kokkos::printf` or we do nothing.

Reply:
I agree with @dalg24, both because of `printf("Hello");` arguably calling the wrong function and in general we should not pollute the global namespace.

Reply:
What does it matter if it calls the wrong function if does the right thing? 
I agree that we should avoid polluting the global namespace but this is as good as we can get it for the `SYCL` backend (unless we can convince the `SYCL` working group to support `printf` in the global namespace which appears to be very unlikely). From a user's perspective, this is a much better solution (for an experimental backend) than a `KOKKOS_IMPL_DO_NOT_USE_PRINTF` macro (since they don't need to change their code at all).

In the end, the question is if we tolerate polluting the global namespace for this backend rather than using `KOKKOS_IMPL_DO_NOT_USE_PRINTF` indefinitely or until we provide `Kokkos::printf` (which would require users to change their code once again).

Reply:
Consider this
```C++
// MyLib_DoWork.hpp
#pragma once
namespace MyLib {
KOKKOS_INLINE_FUNCTION void do_work() { /*...*/ printf("whatever"); }
}
```
Are you OK with that code having different meaning based on transitive header includes?

This opens the door to ODR violations (whether `<Kokkos_Core.hpp>` is included before that header change the definition of `MyLib::do_work`)

Reply:
This file is included by `Kokkos_Macros.hpp`. As soon as you use `KOKKOS_INLINE_FUNCTION`, the imported function is visible.

Reply:
> This file is included by `Kokkos_Macros.hpp`

You are right, I missed that.